pkg/clusteroperator/clusteroperators.go

```diff
@@ -0,0 +1,273 @@
+package clusteroperator
+
+import (
+	"context"
+	"time"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	v1 "github.com/openshift/api/config/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"fmt"
+)
+
+// CoListBuilder provides struct for ClusterOperator List.
+type CoListBuilder struct {
+	Definition *v1.ClusterOperatorList
+	ObjectList *v1.ClusterOperatorList
+	apiClient  *clients.Settings
+	errorMsg   string
+}
+
+// Discover method gets the ClusterOperator in cluster and stores them in the builder struct.
+func (builder *CoListBuilder) Discover() error {
+	glog.V(100).Infof("Getting the list of ClusterOperator objects on this cluster")
+
+	var (
+		coList *v1.ClusterOperatorList
+		err    error
+	)
+
+	coList, err = builder.apiClient.ClusterOperators().List(
+		context.TODO(), metaV1.ListOptions{})
+
+	if err != nil {
+		glog.V(100).Infof("Error to list ClusterOperators")
+
+		return err
+	}
+
+	if len(coList.Items) < 1 {
+		glog.V(100).Infof("Cluster doesn't have ClusterOperator deployed, bad cluster state ...")
+
+		return fmt.Errorf("ClusterOperator list is empty")
+	}
+
+	builder.ObjectList = coList
+
+	for _, co := range coList.Items {
+		glog.V(100).Infof("builder Discover() ClusterOperator contents: %v", co.ObjectMeta.Name)
+	}
+
+	return err
+}
+
+// WaitForAllAvailable waits for all ClusterConfigs to be available for a time duration up to the timeout.
+func (builder *CoListBuilder) WaitForAllAvailable(availableDuration time.Duration,
```

Comment:
Can you re-name it to WaitForAllAvaliableFor? I believe it would be more informative because we are going to wait close function only after **availableDuration** is achieved. So we Wait **For**  a specific amount of time.

Reply:
done

Reply:
done